retriever/lib/scripts.py

```diff
@@ -1,36 +1,71 @@
+
+from future import standard_library
+
+standard_library.install_aliases()
 import csv
 import imp
 import io
 import os
 import sys
-from os.path import join, isfile, getmtime, exists
+import urllib.request
+import urllib.parse
+import urllib.error
+from os.path import join, isfile, getmtime, exists, abspath
 
 from pkg_resources import parse_version
 
-from retriever.lib.compile import compile_json
-from retriever.lib.defaults import SCRIPT_SEARCH_PATHS, VERSION, ENCODING
+from retriever.lib.defaults import SCRIPT_SEARCH_PATHS, VERSION, ENCODING, SCRIPT_WRITE_PATH
+from retriever.lib.load_json import read_json
+
+
+def check_retriever_minimum_version(module):
+    mod_ver = module.retriever_minimum_version
+    m = module.name
+
+    if hasattr(module, "retriever_minimum_version"):
+        if not parse_version(VERSION) >= parse_version("{}".format(mod_ver)):
+            print("{} is supported by Retriever version ""{}".format(m, mod_ver))
+            print("Current version is {}".format(VERSION))
+            return False
+    return True
 
 
 def MODULE_LIST(force_compile=False):
     """Load scripts from scripts directory and return list of modules."""
     modules = []
     loaded_scripts = []
+    if not os.path.isdir(SCRIPT_WRITE_PATH):
+        os.makedirs(SCRIPT_WRITE_PATH)
 
     for search_path in [search_path for search_path in SCRIPT_SEARCH_PATHS if exists(search_path)]:
-        to_compile = [
+        data_packages = [
             file for file in os.listdir(search_path) if file[-5:] == ".json" and
             file[0] != "_" and (
                 (not isfile(join(search_path, file[:-5] + '.py'))) or (
                     isfile(join(search_path, file[:-5] + '.py')) and (
                         getmtime(join(search_path, file[:-5] + '.py')) < getmtime(
                             join(search_path, file)))) or force_compile)]
-        for script in to_compile:
+
+        for script in data_packages:
             script_name = '.'.join(script.split('.')[:-1])
-            compile_json(join(search_path, script_name))
+            if script_name not in loaded_scripts:
+                read_script = read_json(join(search_path, script_name))
+                if read_script:
+                    if not check_retriever_minimum_version(read_script):
+                        continue
+                    setattr(read_script, "_file", os.path.join(search_path, script))
+                    setattr(read_script, "_name", script_name)
+                    modules.append(read_script)
+                    loaded_scripts.append(script_name)
 
         files = [file for file in os.listdir(search_path)
                  if file[-3:] == ".py" and file[0] != "_" and
-                 '#retriever' in ' '.join(open(join(search_path, file), 'r').readlines()[:2]).lower()]
+                 ('#retriever' in
+                  ' '.join(open(join(search_path, file), 'r').readlines()[:2]).lower()
+                  or '# retriever' in
```

Comment:
Is there a reason to add this special case instead of just making all of the comments consistent?